### PR TITLE
Make ignoring compatible with syncthing 0.13 (fixes #110)

### LINKED ISSUE #110: Make handling of ignores in syncthing-notify compatible with syncthing 0.13
My patch that prevents notify from aborting on ignored folders does not work anymore. I traced the issue to the getIgnorePatterns function. It returns without an error but the return value is an empty array. Already the ignores["patterns"] array is empty. As I do not know json or the syncthing http API I stopped tracing the problem there and would very much appreciate it if somebody might take it up from there. Maybe something relevant changed in syncthing API that was not prominently announced in the changelog?

### PR DESCRIPTION
**Pull request by imsodin**
This is NOT COMPATIBLE WITH 0.12! Uses syncthing/lib/ignore package to handle ignoring.

### CODE REVIEW

File: syncwatcher.go [hunk 1]
```diff
@@ -422,26 +380,21 @@ func getFolders() []FolderConfiguration {
...
-}
-
 // watchFolder installs inotify watcher for a folder, launches
 // goroutine which receives changed items. It never exits.
 func watchFolder(folder FolderConfiguration, stInput chan STEvent) {
 	folderPath := expandTilde(folder.Path)
-	ignorePatterns := getIgnorePatterns(folder.ID)
+	ignores := ignore.New(false)
+	Trace.Println("Getting ignore patterns for " + folder.ID)
+	ignores.Load(folderPath + ".stignore")
```
Comment (AudriusButkevicius) [inline syncwatcher.go]: path.Join()

Comment (imsodin) [inline syncwatcher.go]: Thanks, fixed.

**Comment by AudriusButkevicius** [comment]
Looks sane, yet I can't comment on loading the patterns by ourselves.

File: syncwatcher.go [hunk 2]
```diff
@@ -422,26 +380,21 @@ func getFolders() []FolderConfiguration {
...
-}
-
 // watchFolder installs inotify watcher for a folder, launches
 // goroutine which receives changed items. It never exits.
 func watchFolder(folder FolderConfiguration, stInput chan STEvent) {
 	folderPath := expandTilde(folder.Path)
-	ignorePatterns := getIgnorePatterns(folder.ID)
+	ignores := ignore.New(false)
+	Trace.Println("Getting ignore patterns for " + folder.ID)
+	ignores.Load(filepath.Join(folderPath, ".stignore"))
```
Comment (Zillode) [inline syncwatcher.go]: This means we don't recursively load included files (e.g. `#include .stignoreglobal`) :/
Is the ignore rest API fully deprecated?

**Pull request merged**